[compiler-rt] Cleanup use of COMPILER_RT_INCLUDE_TESTS

vitalybuka opened the pull request:
1. Move checks into parent test/CMakeLists.txt
2. COMPILER_RT_INCLUDE_TESTS disable both lit and
   gtests. Before it was very inconsistent between
   sanitizers.

MaskRay reviewed (approved):
I am not familiar with how the tests cmake work, but this patch indeed improves consistency.

For example, `foreach(arch ${MEMPROF_TEST_ARCH})` is at the toplevel while its unittest `foreach` is guarded by `COMPILER_RT_INCLUDE_TESTS`. Removing `COMPILER_RT_INCLUDE_TESTS` is great.

vitalybuka committed:
[compiler-rt] Cleanup use of COMPILER_RT_INCLUDE_TESTS

1. Move checks into parent test/CMakeLists.txt
2. COMPILER_RT_INCLUDE_TESTS disable both lit and
   gtests. Before it was very inconsistent between
   sanitizers.

Reviewers: petrhosek, MaskRay

Reviewed By: MaskRay

Pull Request: https://github.com/llvm/llvm-project/pull/98246
